JetMETCorrections/Modules/plugins/METCorrectorDBReader.cc

```diff
@@ -0,0 +1,143 @@
+  bool mCreateTextFile,mPrintScreen;
+};
+
+}
+
+
+METCorrectorDBReader::METCorrectorDBReader(const edm::ParameterSet& iConfig)
+{
+  mPayloadName    = iConfig.getUntrackedParameter<std::string>("payloadName");
+  mGlobalTag      = iConfig.getUntrackedParameter<std::string>("globalTag");  
+  mPrintScreen    = iConfig.getUntrackedParameter<bool>("printScreen");
+  mCreateTextFile = iConfig.getUntrackedParameter<bool>("createTextFile");
+}
+
+
+METCorrectorDBReader::~METCorrectorDBReader()
+{
+ 
+}
+
+void METCorrectorDBReader::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  edm::ESHandle<METCorrectorParametersCollection> METCorParamsColl;
+  std::cout <<"Inspecting MET payload with label: "<< mPayloadName <<std::endl;
+  iSetup.get<METCorrectionsRecord>().get(mPayloadName,METCorParamsColl);
+
+  // get the sections from Collection (pair of section and METCorr.Par class)
+  std::vector<METCorrectorParametersCollection::key_type> keys;
+  // save level to keys for each METParameter in METParameter collection
+  METCorParamsColl->validKeys( keys );
+  //METCorParamsColl->validSections( sections );
+  for ( std::vector<METCorrectorParametersCollection::key_type>::const_iterator ibegin = keys.begin(),
+	  iend = keys.end(), ikey = ibegin; ikey != iend; ++ikey ) {
+    std::cout<<"--------------------------------------" << std::endl;
+    std::cout<<"Processing key = " << *ikey << std::endl;
+    std::string sectionName= METCorParamsColl->findLabel(*ikey);
+    std::cout<<"object label: "<<sectionName<<std::endl;
+    METCorrectorParameters const & METCorParams = (*METCorParamsColl)[*ikey];
+
+    if (mCreateTextFile)
+    {
+        if(METCorParamsColl->isXYshiftMC(*ikey) )
+        {
+	  std::string outFileName(mGlobalTag+"_XYshiftMC_"+mPayloadName+".txt");
+	  std::cout<<"outFileName: "<<outFileName<<std::endl;
+          METCorParams.printFile(outFileName, sectionName);
+        }else if(METCorParamsColl->isXYshiftDY(*ikey) )
+        {
+	  std::string outFileName(mGlobalTag+"_XYshiftDY_"+mPayloadName+".txt");
+	  std::cout<<"outFileName: "<<outFileName<<std::endl;
+          METCorParams.printFile(outFileName, sectionName);
+        }else if(METCorParamsColl->isXYshiftTTJets(*ikey) )
+        {
+	  std::string outFileName(mGlobalTag+"_XYshiftTTJets_"+mPayloadName+".txt");
+	  std::cout<<"outFileName: "<<outFileName<<std::endl;
+          METCorParams.printFile(outFileName, sectionName);
+        }else if(METCorParamsColl->isXYshiftWJets(*ikey) )
+        {
+	  std::string outFileName(mGlobalTag+"_XYshiftWJets_"+mPayloadName+".txt");
+	  std::cout<<"outFileName: "<<outFileName<<std::endl;
+          METCorParams.printFile(outFileName, sectionName);
+        }else if(METCorParamsColl->isXYshiftData(*ikey) )
+        {
+	  std::string outFileName(mGlobalTag+"_XYshiftData_"+mPayloadName+".txt");
+	  std::cout<<"outFileName: "<<outFileName<<std::endl;
+          METCorParams.printFile(outFileName, sectionName);
```

Comment:
Code is duplicated five times in this if-else statement. It can be condensed to eliminate the duplication:
```
std::string outFileName(mGlobalTag + "_XYshift");
std::string shiftType("MC");
if (METCorParamsColl->isXYshiftDY(*ikey) ) {
	 shiftType = "DY";
} else if (METCorParamsColl->isXYshiftTTJets(*ikey) ) {
	 shiftType = "TTJets";
} else if (METCorParamsColl->isXYshiftWJets(*ikey) ) {
	 shiftType = "WJets";
} else if (METCorParamsColl->isXYshiftData(*ikey) )
           shiftType = "Data";
} else if ( ! METCorParamsColl->isXYshiftMC(*ikey) ) {
         throw cms::Exception("InvalidKey") <<  "************** Can't interpret the stored key: "<<*ikey<<std::endl;
}
outFileName += "_" + mPayloadName + ".txt";
LogDebug("METCorrectorDBReader::analyze") << "outFileName: "<<outFileName<<std::endl;
METCorParams.printFile(outFileName, sectionName);
```